### PR TITLE
LinkRot python script

### PR DESCRIPTION
**Pull request by Polaris000**
README to be updated, the two csv files are the results of a sample blog site I crawled: https://ashiknesin.com/blog/ 
Pass url as command line argument

### CODE REVIEW

File: LinkRot/finder.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import requests
+from bs4 import BeautifulSoup
+import tldextract as t
```
Comment (Pro-Panda) [inline LinkRot/finder.py]: Don't use this library. Acheve the same functionality yourself. Make functions for it

File: LinkRot/finder.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+broken_links = []
+checked_links = []
+count = 0
+
+checked_file = open("checked_urls.csv", "w")
+broken_file = open("broken_urls.csv", "w")
+
+checked_file.write("Sr_No." + "," + "url" + "," + "status_code" + "," + "Proper" + "\n")
```
Comment (Pro-Panda) [inline LinkRot/finder.py]: Make the string to write as a new variable and write the variable. Looks readable

File: LinkRot/finder.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		broken_file.write(url + "," + str(url_request.status_code) + "\n")
+
+
+	print(url_request.status_code)
+	soup = BeautifulSoup(url_request.content, "html.parser")
+
+	url_list = soup.find_all('a', href=True)
+	checked_links.append(url)
+
+	checked_file.write(str(count) + "," + url + "," + str(url_request.status_code) + "," + str(is_ok) + "\n")
```
Comment (Pro-Panda) [inline LinkRot/finder.py]: Make a new variable to write

File: LinkRot/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+bs4==0.0.1
+tldextract==2.2.0
```
Comment (Pro-Panda) [inline LinkRot/requirements.txt]: Remove dependancy

File: LinkRot/broken_urls.csv [hunk 5]
```diff
@@ -0,0 +1 @@
+url,status code
```
Comment (Pro-Panda) [inline LinkRot/broken_urls.csv]: Look for a webpage which has broken links, in order to test your script

File: LinkRot/checked_urls.csv [hunk 6]
```diff
@@ -0,0 +1,8 @@
+Sr No.,url,status code,Proper
```
Comment (Pro-Panda) [inline LinkRot/checked_urls.csv]: No need to add both the .csv files in the script. They are create at runtime anyways

File: LinkRot/finder.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+	for link in url_list:
+
+		new_url = t.extract(link['href'])
+
+		if new_url.domain == url_domain and link['href'] not in checked_links:
+			print(link['href'])
+			read_url(link['href'])
+
+	else:
```
Comment (Pro-Panda) [inline LinkRot/finder.py]: Redundant lines. Why is there a `else: pass` ?

File: LinkRot/finder.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+	print(url_request.status_code)
+	soup = BeautifulSoup(url_request.content, "html.parser")
+
+	url_list = soup.find_all('a', href=True)
+	checked_links.append(url)
+
+	checked_file.write(str(count) + "," + url + "," + str(url_request.status_code) + "," + str(is_ok) + "\n")
+
+	for link in url_list:
+
```
Comment (Pro-Panda) [inline LinkRot/finder.py]: Lots of blank lines which are not required, please remove

Comment (Pro-Panda) [reply LinkRot/finder.py]: use `flake8` fixes
```
sudo apt install flake8
flake8 *.py
```

**Comment by Pro-Panda** [comment]
Please make the changes

**Comment by Pro-Panda** [comment]
Also, I don't think this works. There are more than 100 of links on the homepage itself. But you have checked <10 URLs.
How should it work:
 - Check if all the links present on the page return `200` irrespective of their domain name
 - Go into one of the link and check if the link has the same domain name as the initial site
   - If yes, check all of its links and so on.
   - If no, do nothing

File: LinkRot/finder.py (hunk 1 above)
Comment (Pro-Panda) [reply LinkRot/finder.py]: this == tldextract

File: LinkRot/requirements.txt (hunk 4 above)
Comment (Pro-Panda) [reply LinkRot/requirements.txt]: (of tldextract)

**Comment by Polaris000** [comment]
It works... the requests module just takes too long...(a few seconds per url). I did a keyboard interrupt after a few urls to save time...

Will make the requested changes.. thanx for the feedback

File: LinkRot/broken_urls.csv (hunk 5 above)
Comment (Polaris000) [reply LinkRot/broken_urls.csv]: the sample url I used has one broken link -- you can find it in the broken_urls.csv

File: LinkRot/checked_urls.csv (hunk 6 above)
Comment (Polaris000) [reply LinkRot/checked_urls.csv]: OK, cool  --- they were just to show you how it works, I'll remove them

File: LinkRot/finder.py (hunk 7 above)
Comment (Polaris000) [reply LinkRot/finder.py]: I was checking what happens if there for loop has no iterations from a particular url...
I forgot to remove it

**Comment by Pro-Panda** [comment]
Any updates @Polaris000 ?

**Pull request closed without merging**